pkg/controller/controller.go

```diff
@@ -35,13 +38,85 @@ const queueNumRetries = 3
 type Controller struct {
 	configStoreCache     model.ConfigStoreCache
 	crcController        *onboarding.Controller
+	processor            *processor.Controller
 	serviceIndexInformer cache.SharedIndexInformer
 	adIndexInformer      cache.SharedIndexInformer
 	rbacProvider         rbac.Provider
 	queue                workqueue.RateLimitingInterface
 	adResyncInterval     time.Duration
 }
 
+// convertSliceToMap converts the input model.Config slice into a map with (type/namespace/name) formatted key
+func convertSliceToMap(in []model.Config) map[string]model.Config {
+	out := make(map[string]model.Config)
+	for _, c := range in {
+		key := c.Key()
+		out[key] = c
+	}
+	return out
+}
+
+// equal compares the Spec of two model.Config items
+func equal(c1, c2 model.Config) bool {
+	return c1.Key() == c2.Key() && proto.Equal(c1.Spec, c2.Spec)
+}
+
+// computeChangeList determines a list of change operations to convert the current state of model.Config items into the
+// desired state of model.Config items in the following manner:
+// 1. Converts the current and desired slices into a map for quick lookup
+// 2. Loops through the desired slice of items and identifies items that need to be created/updated
+// 3. Loops through the current slice of items and identifies items that need to be deleted
+func (c *Controller) computeChangeList(current []model.Config, desired []model.Config) []*processor.Item {
```

Comment:
Does this need to be part of the controller object? Maybe if we replace the nil error handlers we will need to have it to utilize some object?